Add required_ruby_version to gemspec

koic opened the pull request:
Follow up of https://github.com/rails/arel/pull/494#issuecomment-321524575.

Arel tested Ruby 2.0.0 or higher with Travis CI, so I wrote that version to `required_ruby_version`.

yahonda commented:
I understand this pull request has been opened to follow up #494 that Arel does not need encoding magic comments. 

I'm wondering which version specified here. At first I was thinking about `">= 2.2.2"` as Rails set [here](https://github.com/rails/rails/blob/master/rails.gemspec#L12) But Rails depends on Arel but Arel does not depends on Rails (who uses Arel without Rails or ActiveRecord?) and Travis CI is still testing Arel with Ruby 2.0.0. 

I'd like to hear from Rails core team members since version requirement is the one of the very important project decisions.

matthewd commented:
Yeah, I'm rather inclined to go to 2.2... partly because it seems rare to use Arel without AR, but mostly just because earlier versions are unsupported anyway.

koic commented:
I see. I think that it is reasonable to specify `">= 2.2.2"` to `required_ruby_version` of arel.gemspec. Because I also think that it is rare to use Arel independently from AR.
I will update this PR like that if it is OK.

Also, I'd like to do maintenance of travis.yml with another PR.
What do you think?

matthewd commented:
Yeah, let's go ahead and do that. We can drop the older rubies from travis in this PR too.

koic commented:
Thanks for replying. I taken your advice. Maintenance of travis.yml was this PR.

travis.yml made it follow Rails (AR). The summary is as follows.

- Drop MRI 2.2.1 or lower and updated with latest MRI versions.
- Drop rbx-2. This build has failed since before.
- Update JRuby version to 9.1.12.0

I updated this PR as described above.
